src/modules/ipfs/client.ts

```diff
@@ -0,0 +1,36 @@
+import { Files } from 'formidable';
+import { PinFileResponse } from './types';
+import uploadFile from './nft.storage';
+import { fromDwebLink } from '.';
+
+export default async function UploadFiles(formFiles: Files ) {
+  const files = Object.values(formFiles)
+    .map(maybeFiles =>  maybeFiles instanceof Array ? maybeFiles[0] : maybeFiles)
+
+  const uploadPromises: Promise<any>[] = []
+  files.forEach((file) => {
+    uploadPromises.push(uploadFile(file.path));
+  })
+
+  const results = await Promise.allSettled(uploadPromises)
```

Comment:
allSettled first time seeing that one. 

```
The Promise.allSettled() method returns a promise that resolves after all of the given promises have either fulfilled or rejected, with an array of objects that each describes the outcome of each promise.
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled